vsphere/internal/virtualdevice/virtual_machine_cdrom_subresource.go

```diff
@@ -472,27 +508,39 @@ func (r *CdromSubresource) Update(l object.VirtualDeviceList) ([]types.BaseVirtu
 		return nil, fmt.Errorf("device at %q is not a virtual CDROM device", l.Name(d))
 	}
 
-	// To update, we just re-insert the ISO as per create, and send it as an edit.
 	dsID := r.Get("datastore_id").(string)
 	path := r.Get("path").(string)
-	ds, err := datastore.FromID(r.client, dsID)
+	clientDevice := r.Get("client_device").(bool)
+	err = r.ValidateCdromConfig()
```

Comment:
As per the comment in validation being in `Create`, this needs to be removed in favour of one that happens during diff customization.